app/Filament/Treso/Pages/AnalyseFactures.php

```diff
@@ -2,49 +2,127 @@
 
+use App\Models\CategorieFacture;
 use App\Models\MontantCategorie;
 use App\Models\FactureRecue;
-use Carbon\Carbon;
+use App\Models\Recettes;
 use Filament\Pages\Page;
 use Filament\Forms\Components\DatePicker;
+use Filament\Forms\Components\Select;
+use Filament\Forms\Components\Grid;
 
 class AnalyseFactures extends Page
 {
     protected static ?string $navigationIcon = 'heroicon-o-chart-bar';
-    protected static ?string $title = 'Analyse des Factures Reçues';
+    protected static ?string $title = 'Analyse Générale';
     protected static string $view = 'filament.treso.pages.analyse-factures';
 
     public $date_debut;
     public $date_fin;
-    public $total = 0;
+    public $date_facture = 'date_facturation';
+    public $date_recette = 'date_fin';
     public $totalsByCategory = [];
+    public $totalRecettes = 0;
+    public $totalDepenses = 0;
 
     protected function getFormSchema(): array
     {
         return [
-            DatePicker::make('date_debut')
-                ->label('Date de début')
-                ->required(),
-            DatePicker::make('date_fin')
-                ->label('Date de fin')
-                ->required(),
+            Grid::make()
+                ->schema([
+                    DatePicker::make('date_debut')
+                        ->label('Date de début')
+                        ->required()
+                        ->columnSpan(1),
+    
+                    DatePicker::make('date_fin')
+                        ->label('Date de fin')
+                        ->required()
+                        ->columnSpan(1),
+    
+                    Select::make('date_facture')
+                        ->label('Date pour la facture')
+                        ->options([
+                            'date_facturation' => 'Date de facturation',
+                            'date_paiement_remboursement' => 'Date de paiement ou remboursement',
+                        ])
+                        ->required()
+                        ->columnSpan(1), 
+    
+                    Select::make('date_recette')
+                        ->label('Date pour la recette')
+                        ->options([
+                            'date_debut' => 'Date de début',
+                            'date_fin' => 'Date de fin',
+                        ])
+                        ->required()
+                        ->columnSpan(1),
+                ]),
         ];
-    }
+    }    
 
     public function calculerTotal()
     {
-        $this->total = FactureRecue::whereBetween('date', [$this->date_debut, $this->date_fin])
-            ->sum('prix');
+        $this->totalDepenses = ($this->date_facture === 'date_facturation' ? 
+            FactureRecue::whereBetween('date', [$this->date_debut, $this->date_fin])
+            :
+            FactureRecue::whereBetween('date_paiement', [$this->date_debut, $this->date_fin])
+                ->orWhere('date_remboursement', [$this->date_debut, $this->date_fin])
```

Comment:
```suggestion
                ->orWhereBetween('date_remboursement', [$this->date_debut, $this->date_fin])
```

J'ai déja changer en prod mais orWhere c'est pas la bonne syntaxe